guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/FileInputDemo.scala

```diff
@@ -12,14 +13,35 @@ object FileInputDemo extends AutoDemo with CssView {
     import org.scalajs.dom.File
     import scalatags.JsDom.all._
 
+    import scala.concurrent.ExecutionContext.Implicits.global
+
     val acceptMultipleFiles = Property(true)
     val selectedFiles = SeqProperty.blank[File]
 
     div(
       FileInput(selectedFiles, acceptMultipleFiles)("files"),
       h4("Selected files"),
       ul(repeat(selectedFiles)(file => {
-        li(file.get.name).render
+        val content = Property(Array.empty[Byte])
+
+        FileService.asBytesArray(file.get) foreach { bytes =>
+          content.set(bytes)
+        }
+
+        val name = file.get.name
+        li(showIfElse(content.transform(_.isEmpty))(
+          span(name).render,
+          {
+            val url = FileService.createURL(content.get)
+            val download = a(href := url.value, attr("download") := name)(name)
+            val revoke = a(href := "#", onclick := { () =>
+              content.set(Array.empty[Byte])
+              url.close()
```

Comment:
You could consider doing `content.clearListeners()` if we don't intend it to change anyway.

Reply:
@ddworak I'm using this `content` as `showIfElse(content.transform(_.isEmpty))`. I don't understand how I can move here to `clearListeners()`.